### PR TITLE
Dependency management changes for JVM-SDK libraries.

### PR DESCRIPTION
**Pull request by praveenkumarct**
#### Summary
Dependency scope changes to Avoid JVM-SDK libraries version mismatch between projects.

#### Description
- This will be a breaking change for the users who are already using Sync-Java library.
- Changed dependencies from Implementation to api. (For more info on this: https://tomgregory.com/how-to-use-gradle-api-vs-implementation-dependencies-with-the-java-library-plugin/)
- Updated Documentation.
- Prepare release 7.0.0.

### CODE REVIEW

File: docs/RELEASE_NOTES.md [hunk 1]
```diff
@@ -38,6 +38,44 @@
...
 -->
 
+### 7.0.0 - Aug 23, 2021
+[Commits](https://github.com/commercetools/commercetools-sync-java/compare/6.0.0...7.0.0) |
+[Javadoc](https://commercetools.github.io/commercetools-sync-java/v/7.0.0/) |
+[Jar](https://search.maven.org/artifact/com.commercetools/commercetools-sync-java/7.0.0/jar)
+
+- 🚧 **Breaking Changes** (1)
+  - **Dependency management:** Avoid JVM-SDK libraries version mismatch between projects.
+     You should not explicitly add JVM-SDK dependencies into your project where you are using Sync-Java. Please remove them if you have already added the below dependencies in your project.
```
Comment (lojzatran) [inline docs/RELEASE_NOTES.md:48]: What happens if users do set explicitly the deps? Also should this note be explicitly written in the main README?

**Review by lojzatran — CHANGES_REQUESTED** [review]

File: docs/RELEASE_NOTES.md (hunk 1 above)
Comment (lojzatran) [reply docs/RELEASE_NOTES.md:48]: I see this lib as prerequisite here: https://github.com/commercetools/commercetools-sync-java#prerequisites

Comment (praveenkumarct) [reply docs/RELEASE_NOTES.md]: https://github.com/commercetools/commercetools-sync-java/pull/761/commits/852e9030cdc0b1fbfde9becd9c033140ab178be7

**Review by lojzatran — APPROVED** [review]

**Comment by praveenkumarct** [comment]
> I would use `commercetools JVM SDK` as JVM SDK is not clear for users.

https://github.com/commercetools/commercetools-sync-java/pull/761/commits/0bf149d4ba9849b0a92e5d71eec8b5b531928f17

**Pull request merged**